lib/middlewares/auth.js

```diff
@@ -0,0 +1,111 @@
+import { ExtractJwt, Strategy as JWTStrategy } from 'passport-jwt';
+
+import { Strategy as LocalStrategy } from 'passport-local';
+import Usuario from '../models/usuario';
+import { generarJWT } from '../helpers/jwt';
+import passport from 'passport';
+
+const opts = {
+  jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken('jwt'),
+  secretOrKey: process.env.SECRET_JWT_SEED,
+};
+
+passport.use(
+  'registro',
+  new LocalStrategy(
+    {
+      usernameField: 'dni',
+      passwordField: 'contrasenia',
+      passReqToCallback: true,
+    },
+    async (req, dni, contrasenia, done) => {
+      try {
+        const usuario = req.body;
+
+        const existeDNI = await Usuario.findOne({ where: { dni } });
+
+        if (existeDNI) {
+          return done('El DNI ya está en uso, por favor utilice otro', false);
+        }
```

Comment:
Tanto esta como las demás validaciones deberían ir al modelo. Ver https://sequelize.org/master/manual/validations-and-constraints.html.

Reply:
Además de que conceptualmente no es acá donde deberían estar, es poco óptimo el código: para crear un usuario hace 4 queries a la DB. Si se resuelve con una constraint unique va a hacer una sola, que eventualmente puede fallar.